### PR TITLE
IGames interface

### PR DESCRIPTION
**Pull request by thomaslmbrt**

### CODE REVIEW

File: includes/IGames.hpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class IGames {
+
+public:
+  virtual ~IGames() {};
+
+public:
+    virtual void start() = 0;
+    virtual std::vector<Arcade::Object *> dump() = 0;
+    virtual void Play() = 0;
+    virtual std::vector<Arcade::Object *> tic(std::chrono::milliseconds time) = 0;
```
Comment (Liroo) [inline includes/IGames.hpp]: tick *

File: includes/IGames.hpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+#include <string>
+#include <chrono>
+
+class IGames {
+
+public:
+  virtual ~IGames() {};
+
+public:
+    virtual void start() = 0;
```
Comment (Liroo) [inline includes/IGames.hpp]: why this indentation is different from line 10 ?

File: includes/IGames.hpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#ifndef IGAMES_HPP
+# define IGAMES_HPP
+
+#include <string>
+#include <chrono>
+
+class IGames {
+
+public:
+  virtual ~IGames() {};
```
Comment (andremaxime) [inline includes/IGames.hpp]: () {} ;     ?

File: includes/IGames.hpp [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+public:
+    virtual void start() = 0;
+    virtual std::vector<Arcade::Object *> dump() = 0;
```
Comment (Liroo) [inline includes/IGames.hpp]: How do you know the namespace Arcade ? And the structure Object, you should include it ;)

File: includes/IGames.hpp [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#ifndef IGAMES_HPP
+# define IGAMES_HPP
+
+#include <string>
+#include <chrono>
+
+class IGames {
```
Comment (Liroo) [inline includes/IGames.hpp]: put this class in Arcade namespace

File: includes/IGames.hpp (hunk 3 above)
Comment (Liroo) [inline includes/IGames.hpp]: This is the default destructor of interface, syntax is like:

virtual ~NameOfTheClass() {};

File: includes/IGames.hpp (hunk 1 above)
Comment (Liroo) [inline includes/IGames.hpp]: use const & to get time

**Comment by andremaxime** [comment]
gtg ! :rocket:

File: includes/IGames.hpp [hunk 6]
```diff
@@ -0,0 +1,26 @@
+#ifndef IGAMES_HPP
+# define IGAMES_HPP
+
+# include <string>
+# include <chrono>
+
+# include "Object.hpp"
+
+namespace Arcade {
```
Comment (Liroo) [inline includes/IGames.hpp]: Add doc here

Comment (Liroo) [inline includes/IGames.hpp]: :)

**Pull request merged**